runtime_scan/pkg/scanner/scanner.go

```diff
@@ -258,35 +258,39 @@ func (s *Scanner) initScan() error {
 	return nil
 }
 
-func (s *Scanner) Scan(scanConfig *_config.ScanConfig, scanDone chan struct{}) error {
+func (s *Scanner) Scan(scanConfig *_config.ScanConfig) (chan struct{}, error) {
 	s.Lock()
 	defer s.Unlock()
 
+	// Create channel for caller to use to track doneness
+	done := make(chan struct{})
+
 	s.scanConfig = scanConfig
 	log.WithFields(s.logFields).Infof("Start scanning...")
 	s.progress.Status = _types.ScanInit
 	if err := s.initScan(); err != nil {
 		s.progress.SetStatus(_types.ScanInitFailure)
-		return fmt.Errorf("failed to initiate scan: %v", err)
+		// Close done channel to indicate that the scan is no longer running
+		close(done)
+		return done, fmt.Errorf("failed to initiate scan: %v", err)
```

Comment:
you are returning a closed channel? is that safe?

Reply:
yup

Reply:
same